ql/src/java/org/apache/hadoop/hive/ql/exec/FileSinkOperator.java

```diff
@@ -910,12 +910,12 @@ private void updateDPCounters(final FSPaths fsp, final int filesIdx) {
   }
 
   private void createDpDirCheckSrc(final Path dpStagingPath, final Path dpFinalPath) throws IOException {
-    if (!fs.exists(dpStagingPath) && !fs.exists(dpFinalPath)) {
```

Comment:
I don't fully understand why we had `dpFinalPath` as a parameter to this method; and why was it also checked before creating the `dpStagingPath` => if `dpFinalPath` was already existed the staging dir was not created....I don't know if that was a bug or intentional....

but now this `dpFinalPath` also become unused; so it could be removed from the function signature

Reply:
if we remove the extra argument we get the same method as `createDpDir` - so there must have been a reason for that if in [HIVE-17778](https://github.com/apache/hive/commit/a9f25c0e7ad3f81a9f00f601947a161516e33f1b#diff-75a40b9ea37fdd5cad43bb2690f796ab678e2c7cf92be78f8de150b3f3c0d98dR819) an extra if and these 2 methods were added.

Reply:
Fix should be in "createDpDir" (ref: stacktrace in the ticket). I have corrected this in the latest patch.